ftw/flamegraph/__init__.py

```diff
@@ -4,19 +4,41 @@
 from contextlib import contextmanager
 from ftw.flamegraph.sampler import Sampler
 from urlparse import parse_qs
+from functools import wraps
 
+import logging
 import os
 import shutil
 import sys
 import tempfile
 import time
-import logging
 
 sampler = None
 logger = logging.getLogger("ftw.flamegraph")
 flamegraph_script = os.path.join(os.path.dirname(__file__), 'flamegraph.pl')
 
 
+def flamegraph(open_svg=True, interval=0.001):
+    """The @flamegraph() decorator wraps a function and produces a flamegraph
+    from calls within this function.
+    """
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            with profile(interval, print_summary=True) as sampler:
+                return_value = func(*args, **kwargs)
+
+            svg_file_path = make_svg(sampler)
+            print '@flamegraph(', svg_file_path, ')'
+            if open_svg:
+                os.system('open {}'.format(svg_file_path))
```

Comment:
This is macOS-specific. I would at least expect something that doesn't result in an error on Linux.
There's xdg-open on Linux, but it's only available with an X desktop, which usually isn't installed on servers.
Or maybe just make `open_svg=False` the default?